src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs

```diff
@@ -459,6 +459,15 @@ protected override bool TryGetNameParts(ExpressionSyntax expression, out IList<s
                 }
             }
 
+            if (nameOrMemberAccessExpression.Parent is PropertyPatternSyntax)
+            {
+                var propertyPattern = nameOrMemberAccessExpression.Parent as PropertyPatternSyntax;
+                foreach (var subPattern in propertyPattern.PatternList.SubPatterns)
+                {
+                    generateTypeServiceStateOptions.PropertiesToGenerate.Add(subPattern);
```

Comment:
i'd actually really like to only add the subpattern's name (or 'Left').  However, this was somewhat unfortunatley typed as 'Token' not as IdentifierNameSyntax.  I found that somewhat unfortunate as the parallel with Object initializers actually has a node.  i.e. if you have ```new Foo { X = 10 }```, then ```X``` is an identifier node, not a token.  

Because of this i had to loosen the typing in our code to allow arbitrary syntax nodes, and node just keep it as name nodes.